### PR TITLE
Integration tests. Move EVM tests to the generic framework

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Fix partially https://github.com/centrifuge/centrifuge-chain/issues/1790

Moving these tests to the generic framework:
- Precompile test
- Ethereum transaction test

~~**NOTE**: Blocked by [`polkadot-v1.1.0`](https://github.com/centrifuge/centrifuge-chain/pull/1756)~~ **READY**

### CODE REVIEW

File: runtime/integration-tests/src/generic/cases/ethereum_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				&& to == contract_address
+				&& transaction_hash == t_hash
+				&& exit_reason == ExitReason::Succeed(ExitSucceed::Returned))
+			.then_some(())
+		})
+		.unwrap();
+	});
+}
+
+crate::test_for_runtimes!([development], call);
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/cases/ethereum_transaction.rs:104]: When trying with `altair` or `centrifuge` I obtain a `BadOrigin` error (when I call `pallet_evm::call()` or `pallet_evm::create()`). Do you know what could be the reason?

**Review by cdamian — DISMISSED** [review]
Thank you for cleaning these up @lemunozm!

File: runtime/integration-tests/src/generic/cases/ethereum_transaction.rs (hunk 1 above)
Comment (cdamian) [reply runtime/integration-tests/src/generic/cases/ethereum_transaction.rs:104]: Not from the top of my head unfortunately, no idea what might cause that.

Comment (lemunozm) [reply runtime/integration-tests/src/generic/cases/ethereum_transaction.rs:104]: I guess some constant we use in the test does not match the runtime config... any idea @mustermeiszer?

**Review by wischli — APPROVED** [review]
LGTM! Thanks for moving this

File: runtime/integration-tests/src/generic/cases/ethereum_transaction.rs (hunk 1 above)
Comment (lemunozm) [reply runtime/integration-tests/src/generic/cases/ethereum_transaction.rs:104]: Leaving this for a future PR

**Pull request merged**